web/src/copy_and_paste.js

```diff
@@ -481,6 +481,61 @@ export function paste_handler_converter(paste_html) {
         },
     });
 
+    // We override the original upstream implementation of this rule to turn any
+    // single line code blocks into inline markdown code. Everything else is the same.
+    turndownService.addRule("fencedCodeBlock", {
+        filter(node, options) {
+            return (
+                options.codeBlockStyle === "fenced" &&
+                node.nodeName === "PRE" &&
+                node.firstChild &&
+                node.firstChild.nodeName === "CODE"
+            );
+        },
+
+        replacement(_content, node, options) {
+            let code = node.firstChild.textContent;
+
+            // We convert single line code inside a code block to inline markdown code,
+            // and the code for this is taken from upstream's `code` rule.
+            if (!code.includes("\n")) {
+                if (!code) {
+                    return "";
+                }
+                code = code.replaceAll(/\r?\n|\r/g, " ");
```

Comment:
What does this do? Shouldn't this not match anything because `code` doesn't contain a `\n` already?

Reply:
Thanks for bringing this to my attention! This line is indeed redundant for single line code. Have removed it